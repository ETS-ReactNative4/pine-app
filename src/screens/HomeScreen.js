import React, { Component } from 'react';
import { StatusBar } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Swiper from 'react-native-swiper';

import { sync as syncWallet } from '../actions/bitcoin/wallet';
import TransactionsScreen from './TransactionsScreen';
import CameraScreen from './CameraScreen';
import ReceiveScreen from './ReceiveScreen';

const SYNC_WALLET_INTERVAL = 60 * 1000 * 1; // 1 minute.

@connect()
export default class HomeScreen extends Component {
  static navigationOptions = {
    header: null
  }

  state = {}

  componentDidMount() {
    const dispatch = this.props.dispatch;

    StatusBar.setBarStyle('dark-content');

    // Sync wallet with an interval.
    this._syncInterval = setInterval(() => {
      dispatch(syncWallet()).catch(() => {
        // Suppress errors.
      });
    }, SYNC_WALLET_INTERVAL);
  }

  componentWillUnmount() {
    clearInterval(this._syncInterval);
  }

  _onIndexChanged(index) {
    const barStyle = index === 0 ? 'light-content' : 'dark-content';

    StatusBar.setBarStyle(barStyle);

    this.setState({
      activeIndex: index
    });
  }

  render() {
    const showCameraPreview = this.state.activeIndex === 0;

    return (
      <Swiper
        index={1}
        loop={false}
        showsPagination={false}
        onIndexChanged={this._onIndexChanged.bind(this)}
      >
        <CameraScreen showPreview={showCameraPreview} />
        <TransactionsScreen />
        <ReceiveScreen />
      </Swiper>
    );
  }
}

HomeScreen.propTypes = {
  dispatch: PropTypes.func
};
