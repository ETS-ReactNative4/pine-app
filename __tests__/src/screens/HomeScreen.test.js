import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../../../src/screens/HomeScreen';

jest.mock('../../../src/containers/ErrorModalContainer', () => 'ErrorModalContainer');
jest.mock('../../../src/containers/BackHeaderContainer', () => 'BackHeaderContainer');
jest.mock('../../../src/containers/ContactListContainer', () => 'ContactListContainer');
jest.mock('../../../src/screens/ContactsScreen', () => 'ContactsScreen');
jest.mock('../../../src/screens/ReceiveScreen', () => 'ReceiveScreen');
jest.mock('../../../src/screens/CameraScreen', () => 'CameraScreen');

const storeMock = {
  getState: jest.fn(() => ({
    network: {
      internet: {
        disconnected: false
      }
    },
    navigate: {
      homeScreen: {
        index: 1
      }
    }
  })),
  dispatch: jest.fn(() => Promise.resolve()),
  subscribe: jest.fn()
};

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <HomeScreen store={storeMock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
