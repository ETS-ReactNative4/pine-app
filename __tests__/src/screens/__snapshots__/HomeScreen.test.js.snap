// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    contentOffset={
      Object {
        "x": 750,
      }
    }
    disableNextButton={false}
    horizontal={true}
    index={1}
    loadMinimal={false}
    loadMinimalSize={1}
    loop={false}
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={false}
    showsVerticalScrollIndicator={false}
    style={undefined}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "flex": 1,
                "justifyContent": "center",
                "padding": 40,
              },
              Object {
                "backgroundColor": "black",
                "padding": 0,
              },
            ]
          }
        >
          <BackHeaderContainer
            backButtonIconStyle={undefined}
            title={undefined}
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              notAuthorizedView={
                <Link
                  onPress={[Function]}
                >
                  Enable Camera Access
                </Link>
              }
              onCameraReady={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#000000",
                  "flex": 1,
                  "height": 1334,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 750,
                }
              }
            />
          </View>
          <ErrorModalContainer />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <TransactionsScreen />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <ReceiveScreen />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
